### ISSUE TITLE
Tests fail since May 3rd because of hardcoded date

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/plone.app.caching#127]
The Jenkins tests [fail](https://jenkins.plone.org/job/plone-6.0-python-3.11/739/) since May 3rd.   Sample error:

```
datetime.datetime(2013, 5, 5, 10, 0, tzinfo=<DstTzInfo 'Europe/Vienna' CEST+2:00:00 DST>) not greater than datetime.datetime(2013, 5, 5, 8, 34, 48, tzinfo=tzutc())

  File "/srv/python3.11/lib/python3.11/unittest/case.py", line 57, in testPartExecutor
    yield
  File "/srv/python3.11/lib/python3.11/unittest/case.py", line 623, in run
    self._callTestMethod(testMethod)
  File "/srv/python3.11/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/home/jenkins/workspace/plone-6.0-python-3.11/src/plone.app.caching/plone/app/caching/tests/test_profile_with_caching_proxy.py", line 136, in test_composite_viewsxx
    self.assertGreater(now, dateutil.parser.parse(browser.headers["Expires"]))
```

The `datetime(2013, 5, 5)` comes from the [`stable_now` test method](https://github.com/plone/plone.app.caching/blob/3.1.0/plone/app/caching/tests/test_utils.py#L27-L32).

At first I suspected a Daylight Savings Time problem, but I think the tests always pass during the first four months of the year, and start failing around the beginning of May and will continue to fail the rest of the year.  But in that case I don't see why this did not fail last year in December, so something else must have changed.

Also, I don't see why the Expires header has a date in 2013.  Something in the tests will set this up, but I don't see it.

I will make a PR where I set the `stable_now` to _June_ 5, so we have a month to fix this properly...

**Comment by mauritsvanrees** [comment]
I merged my temporary solution so Jenkins is fixed for now: https://github.com/plone/plone.app.caching/pull/128

As said, this needs a proper fix, otherwise we will have the same problem in June.

**Comment by mauritsvanrees** [comment]
I expect we will hit this problem again around tomorrow, June 3rd.

Ah, I found the problem!

Tests failed initially because we compare an Expires header with a stable date from May 2013.  I did not understand how the Expires header itself could be from 2013: shouldn't this be 2023?  We are not patching `datetime.now` anywhere.  I changed the stable date to June 2013 to temporarily fix it.

But now I see why the Expires header comes from 2013.  In several places in the code we call `getExpiration(0)`.  When you do this, you get an expiration date from [ten years ago](https://github.com/plone/plone.app.caching/blob/3.1.1/plone/app/caching/operations/utils.py#L585-L595)!  And this expiration date is newer than the `stable_now` that we set.

I will fix `test_utils.stable_now` to use a relative date, so it will continue to work, also ten years in the future.  I will do that as part of PR #130.
